janusgraph-hbase/src/test/java/org/janusgraph/HBaseContainer.java

```
@@ -0,0 +1,174 @@
+    private static final String DEFAULT_VERSION = "2.2.7";
+    private static final String DEFAULT_UID = "1000";
+    private static final String DEFAULT_GID = "1000";
+
+    private final HBaseCompat compat;
+
+    private static String getVersion() {
+        String property = System.getProperty("hbase.docker.version");
+        if (StringUtils.isNotEmpty(property))
+            return property;
+        return DEFAULT_VERSION;
+    }
+
+    private static String getUid() {
+        String property = System.getProperty("hbase.docker.uid");
+        if (StringUtils.isNotEmpty(property))
+            return property;
+        return DEFAULT_UID;
+    }
+
+    private static String getGid() {
+        String property = System.getProperty("hbase.docker.gid");
+        if (StringUtils.isNotEmpty(property))
+            return property;
+        return DEFAULT_GID;
+    }
+
+    private static String getTargetDir() {
+        String property = System.getProperty(HBASE_TARGET_DIR);
+        if (StringUtils.isNotEmpty(property))
+            return property;
+        return Paths.get(System.getProperty("user.dir"), "target").toString();
+    }
+
+    public HBaseContainer() {
+        this(false);
+    }
+
+    public HBaseContainer(boolean mountRoot) {
+        super(new ImageFromDockerfile()
+            .withFileFromPath(".", Paths.get("docker"))
+            .withBuildArg("HBASE_VERSION", getVersion())
+            .withBuildArg("HBASE_UID", getUid())
+            .withBuildArg("HBASE_GID", getGid()));
+        addFixedExposedPort(2181, 2181);
+        addFixedExposedPort(16000, 16000);
+        addFixedExposedPort(16010, 16010);
+        addFixedExposedPort(16020, 16020);
+        addFixedExposedPort(16030, 16030);
+
+        if (mountRoot){
+            try {
+                Files.createDirectories(getHBaseRootdir());
+            } catch (IOException e) {
+                logger.warn("failed to create folder", e);
```

I can see you catch this exception and simply log it. Won't it cause the following code to fail? If so, should we not eat this exception here?

---

It is okay if i wrap it into a IllegalStateException?

---

sounds fine to me. Or maybe JanusGraphException which is also a runtime exception.